tests/dialects/test_tsql.py

```diff
@@ -1,11 +1,101 @@
+            "DISABLE SCALEOUTEXECUTION",
+            "IGNORE_NONCLUSTERED_COLUMNSTORE_INDEX",
+            "KEEP PLAN",
+            "KEEPFIXED PLAN",
+            "MAX_GRANT_PERCENT = 5",
+            "MIN_GRANT_PERCENT = 10",
+            "MAXDOP 13",
+            "MAXRECURSION 8",
+            "NO_PERFORMANCE_SPOOL",
+            "OPTIMIZE FOR UNKNOWN",
+            "PARAMETERIZATION SIMPLE",
+            "PARAMETERIZATION FORCED",
+            "QUERYTRACEON 99",
+            "RECOMPILE",
+            "ROBUST PLAN",
+            "USE PLAN N'<xml_plan>'",
+            "LABEL = 'MyLabel'",
+        ]
+
+        possible_statements = [
+            # These should be un-commented once support for the OPTION clause is added for DELETE, MERGE and UPDATE
+            # "DELETE FROM Table1",
+            # "MERGE INTO Locations AS T USING locations_stage AS S ON T.LocationID = S.LocationID WHEN MATCHED THEN UPDATE SET LocationName = S.LocationName",
+            # "UPDATE Customers SET ContactName = 'Alfred Schmidt', City = 'Frankfurt' WHERE CustomerID = 1",
+            "SELECT * FROM Table1",
+            "SELECT * FROM Table1 WHERE id = 2",
+        ]
+
+        # We test at most 2 options - so first do a run with only 1 option, and afterwards do a cartesian product
+        # of all 2 options.
+        for statement, option in itertools.product(possible_statements, possible_options):
+            query = f"{statement} OPTION({option})"
+            result = self.validate_identity(query)
+            options = result.args.get("options")
+            self.assertIsInstance(options, list, f"When parsing query {query}")
+            is_query_options = map(lambda o: isinstance(o, exp.QueryOption), options)
+            self.assertTrue(all(is_query_options), f"When parsing query {query}")
+
+        for statement, option1, option2 in itertools.product(
+            possible_statements, possible_options, possible_options
+        ):
+            query = f"{statement} OPTION({', '.join([option1,option2])})"
+            result = self.validate_identity(query)
+            options = result.args.get("options")
+            self.assertIsInstance(options, list, f"When parsing query {query}")
+            is_query_options = map(lambda o: isinstance(o, exp.QueryOption), options)
+            self.assertTrue(all(is_query_options), f"When parsing query {query}")
+
+        raising_queries = [
+            # Missing parentheses
+            "SELECT * FROM Table1 OPTION HASH GROUP",
+            # Must be followed by 'PLAN"
+            "SELECT * FROM Table1 OPTION(KEEPFIXED)",
+            # Should expect a numeric literal
+            "SELECT * FROM Table1 OPTION(FAST 'abcd')",
+            # Missing commas
+            "SELECT * FROM Table1 OPTION(HASH GROUP HASH GROUP)",
+        ]
+        for query in raising_queries:
+            with self.assertRaises(ParseError, msg=f"When running '{query}'"):
+                self.parse_one(query, read="tsql")
```

Comment:
Whoops, didn't notice this was `self.parse_one` so I went ahead and used the dialect out of habbit haha
```suggestion
                self.parse_one(query)
```